### ISSUE TITLE
Migration from Legacy SQLiteOpenHelper Schema

### ISSUE THREAD

**Issue by FarhanAhmad** [Raizlabs/DBFlow#771]
Hi, This may not be exactly a question, but I was actually moving from SQLiteOpenHelper Class. And want to migrate the data from old schema to new schema in my app. I actually changed my db schema also when i thought to use DBFlow. So Please point me in direction how can i achive this?

Details for better understanding my problem:

App version: 83749
DB version: 123
DB Scheme: Using SQLiteOpenHelper, i have x tables
I used onCreate to create my schema, onUpdate to apply changes in schema

After Integrating DBFlow
App version: 83750
DB version: ???
DB Schema: Used DBFlow annotation techniques to create tables.

Much appreciated in advance for any help.

**Comment by agrosner** [comment]
This is probably not the place to be asking this, but I am confused on what you are asking.

**Comment by FarhanAhmad** [comment]
Hmm, let me rephrase,

Application is already deployed in production and db was created using SQLiteOpenHelper API provided by android framework. 

Now wanted to use DBFlow in the same application, and the next update users will get have DBFlow in it. But in this update, I have altered some of the tables. So to where exactly put the code for following:

- Altering table script
- Moving data from old tables to new altered table.

Makes sense now?

**Comment by trevjonez** [comment]
@FarhanAhmad, My recommendation to you is that you recreate classes in dbflow that will produce database schema that is identical to your existing latest version. The next step is to translate migrations as need be. This can be done with either or a mix of the migration methods.(java classes / sql files)

From that point if the schema is going to change even further you can continue to add migrations OR you can create all new classes for the new schema and write a migration `0` for the new database that will do the transfer from the old database.

This whole process can take some time (I just did it myself) but one big win here is that you can write unit tests that will test that any schema version of the old database can be migrated all the way up through the chain to the latest.

The easiest way to trick DBFlow into opening your old database is by making sure that it gets moved/renamed so that the open call just grabs the old file. From there so long as the schema matches and the migrations are in place you should be able to do anything you need.
